### ISSUE TITLE
HTTP health checks have a fixed format

### ISSUE THREAD

**Issue by nickmiller-wf** [hashicorp/consul#1527]
Based on what we need to do with health check results, consul's default output format causes some problems for us in terms of parsing. Ideally, we'd like to be able to grab the raw JSON output from our service health endpoints but consul wraps it with some meta information like the HTTP URL used.

https://github.com/hashicorp/consul/blob/master/command/agent/check.go#L401

I couldn't find any issue or PR related to this. Would a PR making this configurable be welcomed?

**Comment by slackpad** [comment]
Hi @nickmiller-wf I'm a little torn on this one - the extra information is sometimes useful (especially for error cases) and I'm a little hesitant to add more configuration complexity around this. For your application can you strip off the leading metadata and then parse the rest as JSON?

**Comment by nickmiller-wf** [comment]
@slackpad Parsing out the JSON shouldn't be a huge issue. Only that we'll have a couple different services doing it. I've determined that the easiest way of adding in this configuration would be to stick it on the check definition itself (`HealthCheck` in `./consul/structs/structs.go`) so it could eventually get set as an unexported field on `CheckHttp` in `./command/agent/check.go`

TL;DR I agree that solving this from consul's side would add a fair amount of configuration complexity. I personally don't have the bandwidth to dedicate towards a refactor but if there is an opportunity to resolve this in any upcoming work, I would be greatly appreciative :) Feel free to close this issue, unless you want it around for future reference.

**Comment by slackpad** [comment]
@nickmiller-wf appreciate the extra info. I'll go ahead and close this down but keep it in mind if we're in that area of the code, or if we get more requests for something like this.
